pyrightconfig.json

```diff
@@ -0,0 +1,60 @@
+{
+  "typeCheckingMode": "basic",
+  // Target the oldest supported version in editors
+  "pythonVersion": "3.7",
```

Comment:
You don't want to have this set; this config file overrides flags so the CI python-version will be overridden: https://github.com/mhammond/pywin32/actions/runs/8084830821/job/22091009766?pr=2102#step:3:9

https://github.com/microsoft/pyright/issues/7330

Reply:
I would recommend unsetting this, and then also not sepcifying the version in the action, instead leaving it to the interpreter set up by setup-python.

Reply:
Thanks for pointing out this gotcha. It's been re-openned as a feature request in pyright, but I see the value of adding a warning for it in pyright-action.

In the mean time, I'll update the CIs I work on and put a mental pin in that to change back later if the way it works is changed.

Reply:
> you're already getting it implicitly set via the interpreter you're using which is sufficient IMO.

I don't see the harm either in being extra explicit here, especially in the context of that feature request, where pyright works in unexpected ways with its configs, and we would re-introduce it if the feature request is accepted.

Reply:
Eric is always so fast, this is already released: https://github.com/microsoft/pyright/releases/tag/1.1.352

Reply:
The latest action also doesn't warn anymore (for 1.1.352+), so you should be able to bump that as well!